### ISSUE TITLE
Sketcher: Allow User to Toggle Automatic Grid Sizeing in Sketcher On and Off

### ISSUE THREAD

**Issue by FreeCAD-Bug-Importer** [FreeCAD/FreeCAD#5573]
Issue imported from https://tracker.freecad.org/view.php?id=1271

* **Reporter:** bejant
* **Date submitted:** 10/12/2013
* **FreeCAD version:** 0.13
* **Category:** Feature
* **Status:** confirmed
* **Tags:** grid

# Original report text

Shamelessly copied from the forum topic referenced below:

"I wish there was a feature where you set it so that the grid covers the entireity of the sketcher view, instead of just a bit near the most recent sketch. Basically, when I try and draw something right now the grid goes away except near the object, so if I draw a 1mm square box on a 1mm grid, and then want to draw another box next to it, the grid goes away entirely. What I am having to do now is draw a huge box around everything, just to keep my grid, then draw my sketch, then erase the huge box."

~https://sourceforge.net/apps/phpbb/free-cad/viewtopic.php?f=19&t=4659&start=0&sid=ccf9198913c5b711c7f8c59ce330d136
or~

https://forum.freecadweb.org/viewtopic.php?f=19&t=4659&start=0

# Additional information

Some users may prefer the automatic sizing feature, and if automatic sizing was done to reduce CPU processing time, maybe a toggle box in the Edit Controls area (directly below the Sketcher Solver area)?  A screen grab of this area is attached. 

# Other bug information

* **Priority:** normal
* **Severity:** feature
* **Category:** Feature
* **Updated:** 2/6/2021

# Discussion from Mantis ticket

### Comment by raulshc 2013-11-16 18:19
This issue is related with #0001049 : Sketcher preferences in preferences editor.

---

### Comment by wmayer 2015-03-23 09:42
See also http://forum.freecadweb.org/viewtopic.php?f=3&t=10203&start=10#p82464

---

### Comment by wmayer 2016-12-01 19:41
See also http://forum.freecadweb.org/viewtopic.php?f=13&t=18771
Possible solution: depending on zoom level and screen size compute the number of grids. If zoomed out too much use an upper limit based on resolution.

---

### Comment by Kunda1 2017-03-13 13:51
Mentioned again in https://forum.freecadweb.org/viewtopic.php?f=3&t=21286&p=165096#p165096

---

### Comment by Kunda1 2017-03-21 12:50
Resolved dupe of this ticket (#2734)

---

### Comment by abdullah 2018-10-06 14:52
Solving a different ticket, I have implemented a dynamic grid size enlarging.

It is not exactly what the reporter requests, but maybe it (partly)fulfils the needs.

Basically, the grid is expanded to surpase the biggest object drawn so far (to the next 10 power, so 5 mm => 10 mm, 50mm => 100mm).

Additionally the grid is now always symetric on the two planes. Before it gave odd representations.

This implementation is now in master and in the PPA.

---

### Comment by kevlevrone 2019-05-03 12:47
Open the file.
Create a new sketch, XY plane.
Toggle construction mode On.
Make sure Auto Remove Redundant is Off
Carbon Copy Sketch005 - constraints 21,24 are showed as redundant.
Undo https://goo.gl/KyvnZF
Toggle construction mode Off.
Carbon Copy Sketch005 - no redundant constraints.

**Comment by adrianinsaval** [comment]
this issue can be closed with the new grid functionality.
